### DISCUSSION TITLE
[WiX4] Table definition error (2728) in simple installer

### DISCUSSION THREAD

**Discussion by jan-tosovsky-cz** [wixtoolset/issues#6495] [Questions] [answered]
When my WiX4 based installer is executed, an exception (error code 2728) is thrown. The message in the log file is localized, but it means something like:
_When you installed this package, setup has encountered an unexpected error. This may indicate a problem with this installer. The error code is 2728. The arguments are: Component ,,._

When searching this error code I got unclear explanation: Table definition error.

 I am newbie so I expect problem rather in my script than in the tool. I just include single exe in the installer. (There will be more stuff in the final version.)

```xml
<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs">
    
    <Package
        Codepage="1250"
        Language="1029"
        Manufacturer="drifted.in"
        Name="JetBrowser"
        UpgradeCode="bd815300-19b5-4ce5-b8c6-1b592be2c574"
        Version="1.0.0"
        InstallerVersion="100"
        ProductCode="4700df0f-d076-4dd8-ae14-c4743e938e00">
        
        <SummaryInformation Description="JetBrowser 1.0 Installer" Manufacturer="drifted.in" Codepage="1250" />
        
        <Media Id="1" Cabinet="data.cab" EmbedCab="yes" />
        
        <StandardDirectory Id="ProgramFiles64Folder">
            <Directory Id="drifted.in" Name="drifted.in">
                <Directory Id="INSTALLDIR" Name="jetbrowser">
                    <Component Id="exe" Guid="5d56d614-add6-48d2-ae1f-3508a19d3626">
                        <File Id="jetbrowser" Name="jetbrowser.exe" Source="jetbrowser.exe" KeyPath="yes" />
                    </Component>
                </Directory>
            </Directory>
        </StandardDirectory>
        
    </Package>
    
</Wix>
```
Can you see some problem here?

**Comment by rseanhall** [comment]
A lot of the internals changed in v4. It's possible it's not handling your file correctly. Can you attach it here?

**Comment by robmen** [comment, marked as answer]
Try adding a `Feature` element and a `ComponentRef` as a child to that. It is entirely possible v4 lost the check to ensure every `Component` is mapped to at least one `Feature`. So, if that fixes it, we have a bug in v4.

**Reply by jan-tosovsky-cz** [reply]
I've followed tutorial at https://www.firegiant.com/wix/tutorial/getting-started/the-files-inside/ but was not patient enough in the Feature part. I skipped this assuming this is optional feature. However, there is explicitly stated: "we have to have at least one". My bad.

**Comment by BMurri** [comment]
I would say then that yes, we have a bug.

That component check for at least one feature is missing.

**Reply by BMurri** [reply]
See issue #6499
